testing/matrix-sdk-integration-testing/src/tests/reactions.rs

```diff
@@ -31,86 +31,102 @@ use matrix_sdk::{
 
     let message_position = timeline.items().await.len() - 1;
+
+    let reaction_key = "👍";
     let reaction = Annotation::new(event_id.clone(), reaction_key.into());
 
     // Toggle reaction multiple times
-    for _ in 0..3 {
-        // Add
-        timeline.toggle_reaction(&reaction).await?;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_remote_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-
-        // Redact
-        timeline.toggle_reaction(&reaction).await?;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-
-        // Add, redact, add, redact, add
-        join_all((0..5).map(|_| timeline.toggle_reaction(&reaction)).collect::<Vec<_>>()).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_remote_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-
-        // Redact, add, redact, add
-        join_all((0..4).map(|_| timeline.toggle_reaction(&reaction)).collect::<Vec<_>>()).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_remote_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-
-        // Redact, add, redact, add, redact
-        join_all((0..5).map(|_| timeline.toggle_reaction(&reaction)).collect::<Vec<_>>()).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-
-        // Add, redact, add, redact
-        join_all((0..4).map(|_| timeline.toggle_reaction(&reaction)).collect::<Vec<_>>()).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-        assert_local_added(&mut stream, user_id, &event_id, &reaction, message_position).await;
-        assert_redacted(&mut stream, &event_id, message_position).await;
-    }
+    let all_tests = spawn(async move {
```

Comment:
Any particular reason to spawn the async block as its own task? The timeout could also be applied to the `async [move] {}` block directly.

Reply:
I always forget an async block is a future, good catch.